Sources/Classes/RSDBPersistentManager.m

```diff
@@ -176,6 +144,105 @@ - (int)openEncryptedDB:(NSString *)encryptionKey {
     return executeCode;
 }
 
+- (void)handleWhenEncryptedDBExists:(RSDBEncryption * _Nullable)dbEncryption isEncryptionNeeded:(BOOL)isEncryptionNeeded {
+    if (isEncryptionNeeded) {
+        // open encrypted database with key
+        int code = [self openEncryptedDB:dbEncryption.key];
+        if (code == SQLITE_NOTADB) {
+            // when key is wrong
+            // delete encrypted database; then open new encrypted database
+            // all previous events will be deleted
+            [RSLogger logError:@"RSDBPersistentManager: createDB: Wrong key is provided. Deleting previous encrypted DB and creating a new encrypted DB"];
+            [self closeDB];
+            [RSUtils removeFile:ENCRYPTED_DB_NAME];
+            [self openEncryptedDB:dbEncryption.key];
+        }
+        return;
+    }
+    
+    // when encryption is disabled and no key is provided
```

Comment:
Please explain to me once, what do you mean by, "when encryption is disabled"?
Does it mean that encryption is not needed? It's a bit of confusing as there is no explicit mention of "when encryption is enabled".

Reply:
yes, I mean `when encryption is not needed`.  Updating the comments to avoid confusion

Reply:
Updated the comment to make it more obvious